frontend/src/components/InputItem.tsx

```diff
@@ -1,12 +1,21 @@
+import { useState } from "react";
 import Button from "react-bootstrap/Button";
+import ItemType from '../types/ItemType';
 
 interface InputItemProps {
-    newItemText: string;
-    setNewItemText: any;
-    addItemToList: any;
+    items: ItemType[];
+    setItems: any;
 };
 
-const InputItem = ({newItemText, setNewItemText, addItemToList}:InputItemProps) => {
+const InputItem = ({items, setItems}:InputItemProps) => {
+
+    const [newItemText, setNewItemText] = useState<string>("");
+
+    const addItemToList = () => {
+        setItems([...items, {text: newItemText, idx: items.length, status: "todo"}]);
+        setNewItemText("");
+    };
```

Comment:
How about we move this function outside the component and use it here as a prop? That way, `items` doesn't need to be provided.